File: common/src/main/java/com/example/common/ActivityInstanceManager.java
package com.example.common;

import android.app.Activity;

import java.util.LinkedList;

public class ActivityInstanceManager {

    //声明链表集合 增删快
    private static LinkedList<Activity> activityList = new LinkedList<>();
    //添加activity
    public static void addActivity(Activity instance) {
        activityList.add(instance);
    }
    //删除activity
    public static void removeActivity(Activity instance) {
        if (activityList.contains(instance)) {
            activityList.remove(instance);
        }
    }
    //结束全部activity
    public static void finishAllActivity() {
        for (Activity item : activityList) {
            item.finish();
        }
    }
}

File: common/src/main/java/com/bw/common/ActivityInstanceManager.kt
package com.bw.common

import android.app.Activity

class ActivityInstanceManager {

    //为什么使用MutableList因为进行频繁添加删除操作
    var activity_list:MutableList<Activity> = mutableListOf()

    //添加Activity
    fun addActivity(instance:Activity){
        activity_list.add(instance)
    }

    //删除Activity
    fun removeActivity(instance: Activity){
        if (activity_list.contains(instance)){
            activity_list.remove(instance)
        }
    }

    //关闭应用中所有的打开的Activity
    fun finishAllActivity(){
        for (item in activity_list){
            item.finish()
        }
    }

}